web/src/helpers/zonedata.js

```diff
@@ -1,3 +1,46 @@
+const yaml = require('js-yaml');
+const path = require('path');
+const fs = require('fs');
```

Comment:
Sadly we can't use `fs` in the frontend as it is a `nodejs` specific module.

Any and all "yaml bundling" has to happen in the build step.

Reply:
I am actually not sure how to do this. Any thoughts on how to import all yaml files in the directory?

Reply:
I know how to import the YAML files, that is easy with something like [yaml-loader](https://www.npmjs.com/package/yaml-loader).

The problem I haven't figured out yet is how to combine the yaml files before webpack bundles them. But I think there is a way to run custom scripts as a part of the webpack build system but I think that would break snowpack. (We could just use webpack for local development as well with a separate webpack config instead tho.)

>**Note**
>This would be considerable easier if it was just JSON files as we could skip the whole YAML conversion step.

Reply:
found a workaround, see [#7e58bb23ad2b7c35f9bff8d167752fde27c46104](https://github.com/electricitymap/electricitymap-contrib/pull/4435/commits/7e58bb23ad2b7c35f9bff8d167752fde27c46104)

Reply:
The workaround forces the user to manually run the `yarn generate-zones-config` command, but it's required for local development, so that's the reason I did not extract it to automatically run at build time. Same thing as for the geo world update

Reply:
> > > Windows will not let you put > in the name and honestly I have no idea what will happen if you try and open a file with it.
> > 
> > 
> > Is it because of the `>` or because you didn't provide an extension file?
> > Otherwise I'll find a way around
> 
> It's cause of the `>` it's a reserved character in windows filenames.
> 
> > Use any character in the current code page for a name, including Unicode characters and characters in the extended character set (128–255), except for the following:
> > The following reserved characters:
> 
> ```
> < (less than)
> > (greater than)
> : (colon)
> " (double quote)
> / (forward slash)
> \ (backslash)
> | (vertical bar or pipe)
> ? (question mark)
> * (asterisk)
> ```
> 
> Fun fact: Windows work fine without file extensions when manually open a file with a program (that supports it) like vs code.

I think the easiest would be to just change the exchange filenames from `zoneKey1->zoneKey2` to `zoneKey1_to_zoneKey2`, what do you think?

Reply:
Hmm I am slightly worried about this process. if I understand correctly

1. This would mean all changes to any zone config, eg capacity changes requires the contributor to run yarn etc. Unless we set up some kind of automatic job, this would make it very hard for contributors who do not have a full development setup.
2. We duplicate all zone configs (in single file and the bundled version)
3. During local development, we'd also have to rerun the script all the time if we're working on something related to config.

The geometry process is in my opinion a bit different because it requires a lot of processing and validation and is not something we do that often.

Maybe there's a way for us to avoid bundling it, perhaps there's some kind of configuration within Webpack/snowpack that allows us to import it in separate files?

Reply:
> Hmm I am slightly worried about this process. if I understand correctly
> 
> 1. This would mean all changes to any zone config, eg capacity changes requires the contributor to run yarn etc. Unless we set up some kind of automatic job, this would make it very hard for contributors who do not have a full development setup.
> 2. We duplicate all zone configs (in single file and the bundled version)
> 3. During local development, we'd also have to rerun the script all the time if we're working on something related to config.
> 
> The geometry process is in my opinion a bit different because it requires a lot of processing and validation and is not something we do that often.
> 
> Maybe there's a way for us to avoid bundling it, perhaps there's some kind of configuration within Webpack/snowpack that allows us to import it in separate files?
> 

We could git ignore the merged files and create them locally with a postinstall script.

That way it would be safe to edit them in the browser without adding an extra manual step locally but it would create some extra overhead on local development.

Reply:
Yep was thinking of something similar, I have I can find a good solution to this, stay tuned